### ISSUE TITLE
i try to combine in angular 6 but got this error

### ISSUE THREAD

**Issue by jeffkenz89** [chase-moskal/webp-hero#17]
c:/Angular/wrath/node_modules/webp-hero/libwebp/dist/webp.cjs.js
Module not found: Error: Can't resolve 'fs' in 'c:\Angular\wrath\node_modules\webp-hero\libwebp\dist'

i got this error,i already has fs node module install do you know why this problem occured?

**Comment by chase-moskal** [comment]
@jeffkenz89 — thanks for reporting this error

i believe it is the same error as #5 which was reported by @vijaynt some time ago

investigating the issue further, i've found that the error is originating from the google/emscripten output in `webp-hero/libwebpjs/dist/google/webp.js`, which webp-hero then wraps up into the commonjs module you are using at `webp-hero/libwebpjs/dist/webp.cjs.js`

the error comes from the following code:

```js
if (ENVIRONMENT_IS_NODE) {
    if (!Module["print"]) Module["print"] = console.log;
    if (!Module["printErr"]) Module["printErr"] = console.warn;
    var nodeFS;
    var nodePath;
    Module["read"] = function shell_read(filename, binary) {

        if (!nodeFS) nodeFS = require("fs");
//                            ^^^^^^^^^^^^^
//                 (here is the offending require call)

        if (!nodePath) nodePath = require("path");

```

as we can see, `ENVIRONMENT_IS_NODE` is supposed to be guarding the "fs" require call so that the module will work in any environment

my suspicion is that the angular bundler is sniffing out the `require` calls anyways

**Comment by chase-moskal** [comment]
ahah, here's the ticket: https://github.com/emscripten-core/emscripten/issues/6653

it turns out, the require calls are added by emscripten to detect the environment

then angular's bundler assumes these are dependencies and attempts to roll them in, but doesn't allow "fs" and other node modules anymore

in that emscripten issue, somebody cites a solution:

> One thing you could try is to compile with `-s ENVIRONMENT="web"`. All require statements should then be removed from the output.

i'll look into implementing this soon

**Comment by jeffkenz89** [comment]
tq for the solution will try that later

**Comment by chase-moskal** [comment]
i've attempted a fix, but am running into a confusing error from emscripten

i ask about it in this closed issue: https://github.com/emscripten-core/emscripten/issues/6717#issuecomment-486007470

if nobody answers there soon, i should open a new issue to ask about it in the emscripten repo

**Comment by chase-moskal** [comment]
@jeffkenz89 — fix has been published in `webp-hero@0.0.0-dev.21`

fixed in 4904f32c1957af3ec2f399cb310c3fef8ce725ca

 :tada: :+1:
